junit/src/main/java/cucumber/api/junit/CucumberRule.java

```diff
@@ -0,0 +1,139 @@
+package cucumber.api.junit;
+
+import cucumber.runtime.ClassFinder;
+import cucumber.runtime.Runtime;
+import cucumber.runtime.RuntimeOptions;
+import cucumber.runtime.RuntimeOptionsFactory;
+import cucumber.runtime.io.MultiLoader;
+import cucumber.runtime.io.ResourceLoader;
+import cucumber.runtime.io.ResourceLoaderClassFinder;
+import cucumber.runtime.junit.FeatureRunner;
+import cucumber.runtime.junit.JUnitReporter;
+import cucumber.runtime.model.CucumberFeature;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.Statement;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+
+public class CucumberRule implements TestRule {
+    private final RuntimeOptions runtimeOptions;
+
+    public CucumberRule() {
+        this(new RuntimeOptionsFactory(CucumberRule.class /* no conf */).create());
+    }
+
+    public CucumberRule(final RuntimeOptions runtimeOptions) {
+        this.runtimeOptions = runtimeOptions;
+    }
+
+    public CucumberRule(final String defaultPackage, final String... args) {
+        final List<String> options = new ArrayList<String>(args == null ? Collections.<String>emptyList() : asList(args));
+        if (defaultPackage != null) {
+            if (options.isEmpty()) {
+                options.add(MultiLoader.CLASSPATH_SCHEME + defaultPackage.replace('.', '/'));
+            }
+            if (!hasOption(options, "--glue")) {
+                options.add("--glue");
+                options.add(defaultPackage);
+            }
+            if (!hasOption(options, "--plugin")) {
+                options.add("--plugin");
+                options.add("pretty");
+            }
```

Comment:
Agree but less cucumber like. Maybe a core refactoring to build cli on top of it?